### PR TITLE
Fix empty member removal log

### LINKED ISSUE #27: A Restart Pod may be permantly removed from cluster
**Description**

A Restart Pod may be permantly removed from cluster. This is because the `prestop.sh` script is not atomic when remove itself from cluster & write `member_removal.log`.

see prestop.sh:

``` sh
etcdctl member remove "$(cat "${ETCD_DATA_DIR}/member_id")" "${extra_flags[@]}" > "$(dirname "$ETCD_DATA_DIR")/member_removal.log"
```

Due to network or other problem, the leader may already removed the `member_id`, but the node running `prestop.sh` doesn't receive the result.

**Steps to reproduce the issue:**

1. use helm chart to create etcd cluster with 3 nodes
2. use some traffic trick like `tc` to make lantency long
3. delete the stateful set with --cascade=orphan
4. save a follower node's manifest
5. delete the follower node
6. delete the `tc` rule, to make lantency normal
7. start the follower node again

**Describe the results you received:**

The restarted follower doesn't leave a correct removal log before exit, so it restart with old `member_id`, which is already permantly removed from cluster.

**Describe the results you expected:**

The restarted follower should check if the old `member_id` is removed from cluster, and use a new member_id if needed.

**Additional information you deem important (e.g. issue happens only occasionally):**

image version:  `3.5.0-debian-10-r24`

### PR DESCRIPTION
**Pull request by haorenfsa**
**Description of the change**

ensure memory buffers are flushed to disk for `member_removal.log`

**Applicable issues**

Fix #27  in most situations

**Additional information**

like #21

### CODE REVIEW

**Comment by rafariossaa** [comment]
Hi,
  We are testing this change in our CI/CD.

File: 3/debian-10/rootfs/opt/bitnami/scripts/etcd/prestop.sh [hunk 1]
```diff
@@ -19,4 +19,4 @@ if is_empty_value "${endpoints}"; then
 fi
 read -r -a extra_flags <<< "$(etcdctl_auth_flags)"
 extra_flags+=("--endpoints=${endpoints}" "--debug=true")
-etcdctl member remove "$(cat "${ETCD_DATA_DIR}/member_id")" "${extra_flags[@]}" > "$(dirname "$ETCD_DATA_DIR")/member_removal.log"
+stdbuf -oL etcdctl member remove "$(cat "${ETCD_DATA_DIR}/member_id")" "${extra_flags[@]}" > "$(dirname "$ETCD_DATA_DIR")/member_removal.log"
```
Comment (juan131) [inline 3/debian-10/rootfs/opt/bitnami/scripts/etcd/prestop.sh:22]: I guess you used [this](https://github.com/bitnami/bitnami-docker-etcd/blob/master/3/debian-10/rootfs/opt/bitnami/scripts/libetcd.sh#L318) as reference, right? Could you please copy-paste the same comment (lines 314-317) to ensure we have the context about why we're using stdbuf? thanks in advance

**Review by juan131 — COMMENTED** [review]
Thanks so much for this! We really appreciate it! Your changes look good to me, please check my suggestion.

File: 3/debian-10/rootfs/opt/bitnami/scripts/etcd/prestop.sh (hunk 1 above)
Comment (haorenfsa) [reply 3/debian-10/rootfs/opt/bitnami/scripts/etcd/prestop.sh:25]: Added. The comment is not quite accurate, so I edit it as well

**Review by juan131 — APPROVED** [review]
Thanks so much! LGTM

**Comment by juan131** [comment]
@haorenfsa your changes were included at `3.5.1-debian-10-r61`, see:

- https://github.com/bitnami/bitnami-docker-etcd/commit/a844850585a30b652f89149ae7bbb128361b780b

**Pull request merged**